GliaWidgets/Component/NewMessageIndicator/UnreadMessageIndicatorView.swift

```diff
@@ -0,0 +1,75 @@
+import UIKit
+
+final class UnreadMessageIndicatorView: View {
+    var newItemCount: Int = 0 {
+        didSet {
+            if newItemCount <= 0 {
+                isHidden = true
+            } else {
+                isHidden = false
+                badgeView.newItemCount = newItemCount
+            }
+        }
+    }
+
+    var tapped: (() -> Void)?
+
+    private let style: UnreadMessageIndicatorStyle
+
+    private let backgroundView = UIImageView()
+    private let userImageView: UserImageView
+    private let badgeView: BadgeView
+
+    private let kSize = CGSize(width: 58, height: 66) // Shadows included
+    private let kUserImageSize = CGSize(width: 36, height: 36)
+    private let kBadgeInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 5)
+
+    public init(with style: UnreadMessageIndicatorStyle) {
```

Comment:
Since the `UnreadMessageIndicatorView` is internal, do we need to have this initializer as public?

Reply:
Must've copied from somewhere and didn't notice. Thanks!